RFC-0039-cuda-support.md

```diff
@@ -0,0 +1,196 @@
+
+# [CUDA version support]
+
+**Authors:**
+* @atalman @malfet
+
+## **Motivation**
+
+The proposal is to provide two main benefits
+
+- Aligned decision policy provides transparent decision making;
+- Incorporate decision points in the early release process to stagger CUDA migration from the RC release integration work.
+
+## **Proposed Policy and Process**
+
+### We should introduce new version of CUDA when
+
+- It enables new important GPU architecture (For example A100 with CUDA-11.1)
+- It brings significant performance improvements
+- It fixes significant correctness issues
+- It significantly reduces binary/memory footprint
+
+### We would deprecate version of CUDA when
+
+As soon as we introduce a new Experimental Version we should consider moving the previous Experimental Version to Stable, and decommission the previous Stable version. Typically we want to support 2-3 versions of CUDA as follows:
+
+- Optional Legacy Version: If we need to have 1 version for backend compatibility or to work around the current limitation. For example: CUDA older driver is incompatible with newer CUDA version
+- Stable Version: This is a stable CUDA version that is used most of the time. This is the version we want to upload to PyPI. Please note that if the stable version is equal or slightly older to the version then the version fbcode is using, we probably need to keep it in CI/CD system as Optional Legacy Version.
+- Latest Experimental Version: This is the latest version of CUDA that we want to support. Minimal requirement is to have it available in nightly releases (CD)
+
+### Detailed Process of Introducing new CUDA version
+
+1. Evaluate CUDA update necessity
+  Goal: When Any of this is true:
```

Comment:
Are there examples of cuda version updates that do _not_ meet this criteria?

The hard decisions are going to be when someone wants to argue that a upgrade does NOT meet the bar, so having real-world examples of those will help guide those future discussions

Reply:
Yes, I believe the history for the updates are here: